src/com/amazon/ionschema/internal/IonSchemaSystemLoggerInternal.kt

```diff
@@ -0,0 +1,29 @@
+/*
+ * Copyright 2021 Amazon.com, Inc. or its affiliates. All Rights Reserved.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License").
+ * You may not use this file except in compliance with the License.
+ * A copy of the License is located at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * or in the "license" file accompanying this file. This file is distributed
+ * on an "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either
+ * express or implied. See the License for the specific language governing
+ * permissions and limitations under the License.
+ */
+
+package com.amazon.ionschema.internal
+
+import com.amazon.ionschema.IonSchemaSystemLogger
+import com.amazon.ionschema.LogLevel
+
+/**
+ * Logger that adds "[ion-schema-kotlin]" prefix to all lines and then delegates to
+ * a logging callback.
+ */
+internal class IonSchemaSystemLoggerInternal(private val delegate: IonSchemaSystemLogger) {
+    operator fun invoke(level: LogLevel, message: () -> String) {
+        delegate.invoke(level) { "[ion-schema-kotlin] " + message() }
```

Comment:
I'm on the fence about whether this is valuable. The callback is specific to this library, so users who want this could add the prefix themselves. Or dump a stack trace that identifies the specific classes, or do any number of other things.